### DISCUSSION TITLE
Automatically setting on `read-only` application intent

### DISCUSSION THREAD

**Discussion by humada05** [microsoft/mssql-jdbc#2332] [Q&A] [answered]
Hi,
We have a couple customers that wanted us to enable `read-only` application intent for our connections to their instances. After adding this, we thought this driver setting could apply to other customer databases as well, but we hit issues trying to do that. Specific examples were wtih mirrored instances and CT not being supported for secondaries.

Is it possible for the MSSQL driver to automatically switch between `read-only` and `read-write` application intents? We were intending to create this logic ourselves, but its quite difficult to understand when a database is able to accept `read-only` intent connections.

**Comment by David-Engel** [comment, marked as answer]
Closing this discussion in favor of issue #2333.
